File: lingting-core/src/main/java/live/lingting/framework/value/multi/StringMultiValue.java
package live.lingting.framework.value.multi;

import java.util.ArrayList;
import java.util.Collection;
import java.util.Collections;
import java.util.function.Supplier;

/**
 * @author lingting 2024-09-14 11:21
 */
public class StringMultiValue extends AbstractMultiValue<String, String, Collection<String>> {

	public StringMultiValue() {
		this(ArrayList::new);
	}

	protected StringMultiValue(Supplier<Collection<String>> supplier) {
		super(supplier);
	}

	protected StringMultiValue(boolean allowModify, Supplier<Collection<String>> supplier) {
		super(allowModify, supplier);
	}

	@Override
	public StringMultiValue unmodifiable() {
		StringMultiValue value = new StringMultiValue(false, supplier);
		value.from(this, Collections::unmodifiableCollection);
		return value;
	}

}

File: lingting-core/src/main/kotlin/live/lingting/framework/value/multi/StringMultiValue.kt
package live.lingting.framework.value.multi


import java.util.Collections
import java.util.function.Supplier

/**
 * @author lingting 2024-09-14 11:21
 */
open class StringMultiValue : AbstractMultiValue<String, String, MutableCollection<String>> {
    constructor() : this(Supplier<MutableCollection<String>> { ArrayList() })

    protected constructor(supplier: Supplier<MutableCollection<String>>) : super(supplier)

    protected constructor(allowModify: Boolean, supplier: Supplier<MutableCollection<String>>) : super(allowModify, supplier)

    override fun unmodifiable(): StringMultiValue {
        val value = StringMultiValue(false, supplier)
        value.from(this) { Collections.unmodifiableCollection<String>(it) }
        return value
    }
}
